File: cpp/1st_Week/Praktikum/MainPP.cpp
#include "PersegiPanjang.cpp"
using namespace std;

int main(){
    PersegiPanjang rec1;
    
    //inisialisasi menggunakan setter
    rec1.setPanjang(6);
    rec1.setLebar(3);

    //inisialiasi menggunakan constructor
    PersegiPanjang rec2(10, 5);

    //memakai inputan
    int n = 0, i = 0;
    cin >> n;

    //membuat array object
    PersegiPanjang arrRec[n];

    for ( i = 0; i < n; i++)
    {
        int p, l;
        cin >> p >> l;

        //memasukan nilai array object
        arrRec[i].setLebar(l);
        arrRec[i].setPanjang(p);
    }
    
    cout<<"Keliling Persegi Panjang 1 : "<< rec1.keliling() << endl;
    cout<<"Luas Persegi Panjang 1 : "<< rec1.luas() << endl;
    cout<<"Keliling Persegi Panjang 2 : "<< rec2.keliling() << endl;
    cout<<"Luas Persegi Panjang 2 : "<< rec2.luas() << endl;

    //output array
    for ( i = 0; i < n; i++)
    {
        cout<<"Keliling Persegi Panjang array : " << arrRec[i].keliling() << endl;
        cout<<"Luas Persegi Panjang array : "<< arrRec[i].luas() << endl;
    }
    
    return 0;
}

File: python/1st_week/Praktikum/MainPP.py
import imp
from PersegiPanjang import PersegiPanjang

# memakai setter
rec1 = PersegiPanjang()
rec1.setPanjang(6)
rec1.setLebar(3)

# memakai constructor
rec2 = PersegiPanjang(10,5)

n = int(input())

# jika objek nya banyak, begini cara alokasi class dan instalasi objeknya
arrRec = [PersegiPanjang() for i in range(n)]
i = 0

for i in range(n):
    # cara input banyak angka dalam satu baris
    p, l = map(int, input().split())
    # menampung nilai menggunakan setter
    arrRec[i].setPanjang(p)
    arrRec[i].setLebar(l)

# ambil nilai dengan getter
print("Keliling persegi panjang 1 : " + str(rec1.keliling()))
print("Luas persegi panjang 1 : " + str(rec1.luas()))
print("Keliling persegi panjang 2 : " + str(rec2.keliling()))
print("Luas persegi panjang 2 : " + str(rec2.luas()))

# jika memakai perulangan
i = 0
for i in range(n):
    print(f"keliling array persegi panjang ke {i+1} : {arrRec[i].keliling()}")
    print(f"luas array persegi panjang ke {i+1} : {arrRec[i].luas()}")